images/controller/internal/controllers/rv_status_config_shared_secret/reconciler.go

```diff
@@ -0,0 +1,275 @@
+/*
+Copyright 2025 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rvstatusconfigsharedsecret
+
+import (
+	"context"
+	"slices"
+
+	"github.com/go-logr/logr"
+	"github.com/google/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/deckhouse/sds-replicated-volume/api/v1alpha3"
+)
+
+type Reconciler struct {
+	cl  client.Client
+	log logr.Logger
+}
+
+var _ reconcile.Reconciler = (*Reconciler)(nil)
+
+// NewReconciler creates a new Reconciler instance.
+// This is primarily used for testing, as fields are private.
+func NewReconciler(cl client.Client, log logr.Logger) *Reconciler {
+	return &Reconciler{
+		cl:  cl,
+		log: log,
+	}
+}
+
+func (r *Reconciler) Reconcile(
+	ctx context.Context,
+	req reconcile.Request,
+) (reconcile.Result, error) {
+	log := r.log.WithName("Reconcile").WithValues("req", req)
+	log.Info("Reconciling")
+
+	// Get the RV
+	rv := &v1alpha3.ReplicatedVolume{}
+	if err := r.cl.Get(ctx, req.NamespacedName, rv); err != nil {
+		log.Error(err, "Getting ReplicatedVolume")
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Check if sharedSecret is not set - generate new one
+	// Filtering through early return instead of WithEventFilter
+	if rv.Status == nil || rv.Status.DRBD == nil || rv.Status.DRBD.Config == nil || rv.Status.DRBD.Config.SharedSecret == "" {
+		return r.reconcileGenerateSharedSecret(ctx, rv, log)
+	}
```

Comment:
```suggestion
	// Check if sharedSecret is not set - generate new one
	if rv.Status == nil || rv.Status.DRBD == nil || rv.Status.DRBD.Config == nil || rv.Status.DRBD.Config.SharedSecret == "" {
		return r.reconcileGenerateSharedSecret(ctx, rv, log)
	}
```

Тут фильтрацию надо делать даже с фильтром, потому что это новый Get